client/src/App.js

```diff
@@ -59,23 +63,86 @@ class App extends Component {
         ProxyWalletAbi,
         ProxyWalletAddress,
       );
-
+     
       const FutureTokenAddress = map.dev.FutureToken.toString();
       const FutureTokenAbi = FutureToken.abi;
       const FutureTokenInstance = new web3.eth.Contract(
         FutureTokenAbi,
         FutureTokenAddress,
       );
 
+      // cUSDC address and ABI
+      // I've moved this up so that I can call the cUsdcAddress lower
+      const cUsdcAddress = config.cUsdcAddress;
+      const cUsdcAbi = config.cUsdcAbi;
+      const cUsdcContract = new web3.eth.Contract(
+        cUsdcAbi, 
+        cUsdcAddress,
+      );
+
 
       console.log('proxyWalletInstance: ' + ProxyWalletInstance);
-      const proxyClone = await ProxyWalletInstance.methods.getOrCreateClone().call();
-      const proxyAddress = await ProxyWalletInstance.methods.proxyAddress().call();
-      const proxyCloneAddress = await ProxyWalletInstance.methods.getCloneAddress().call();
+      
+      //Goal is to create a clone contract instance
+      //this transaction gets or creates a clone.
+      //we should check if the userAccount already has a clone deployed
+      //declare a cloneAddress variable
+      var cloneAddress = '';
+      
+      try {
+        //try calling the getClone method. If there is no clone, then we will get an error
+        cloneAddress = await ProxyWalletInstance.methods.getClone().call({'from': userAccounts[0]});
+      }
+        catch (error){
+        //on the error, ask user to send a transaction creating a clone and console print the cloneAddress
+        cloneAddress= await ProxyWalletInstance.methods.getOrCreateClone().send({'from': userAccounts[0]});
+      }
```

Comment:
Good effort, In javascript it is usual to check for the existence of the object and if it is not present then create.
   cloneAddress = await ProxyWalletInstance.methods.getClone().call({'from': userAccounts[0]});
   if (!cloneAddress) {  
   cloneAddress = await ProxyWalletInstance.methods.getClone().call({'from': userAccounts[0]});
   }